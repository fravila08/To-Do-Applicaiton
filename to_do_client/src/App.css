body {
  width: 100vw;
  height: 100vh;
}

.App {
  width: 100vw;
  height: 100vh;
  margin: 0;
}

h1 {
  color: black;
  font-weight: 300;
  text-shadow: 0em;
  text-decoration: underline;
}

#Header {
  text-align: center;
}

/* List holders [pending/completed] */

.listHolder {
  border: black solid 4px;
  min-height: 40vh;
  background-color: lightgrey;
  text-align: center;
  padding: 0 5vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: auto;
  height: 10vh;
}

#CompletedList {
  border-radius: 0 0 10vw 10vw;
}

#pending {
  border-radius: 10vw 10vw 0 0;
}

.listHolder::-webkit-scrollbar {
  display: none;
}

.listHeader {
  background-color: whitesmoke;
  border-radius: 5vw;
  margin-top: 1vh;
  padding: 1vh 10vw;
  text-align: center;
  width: 50vw;
}

/* Task Row */

.task {
  background-color: grey;
  border: black 2px solid;
  height: 4vh;
  width: 80vw;
  border-radius: 1vh;
}

.taskTitle {
  display: flex;
  text-align: left;
  font-size: 2vh;
}

.checkHolder {
  display: flex;
  justify-content: center;
  align-items: top;
  position: relative;
}
.check {
  position: absolute;
  opacity: 0;
  z-index: 2;
}

.checkImg {
  height: 3vh;
  position: absolute;
  z-index: 1;
}

/* Input field on upper right hand corner */

.FormHolder {
  margin: 0;
  padding: 0;
}

Form {
  display: flex;
  flex-direction: row-reverse;
}

#createTaskInput {
  font-size: 2vh;
  max-width: 50vw;
}

#createTaskButton {
  border-radius: 0 1vh 1vh 0;
  height: 5vh;
  position: absolute;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}
