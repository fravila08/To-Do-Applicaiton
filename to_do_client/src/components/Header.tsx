import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import React, { useEffect, useState } from "react";
import axios from "axios";
import { ITask } from "../App";

interface HeaderProps {
  allTasks: ITask[];
  setAllTasks: (allTasks: ITask[]) => void;
}

export interface ResponseCreateTask {
  itemCreated: boolean;
  id: number;
}

export const createTask = async (taskTitle: string): Promise<ResponseCreateTask> => {
  let response = await axios.post("newtask/", {
    name: taskTitle,
  });
  return response["data"];
};

export const Header: React.FC<HeaderProps> = ({ allTasks, setAllTasks }) => {
  const [isDisabled, setIsDisabled] = useState(true);
  const [newTask, setNewTask] = useState("");

  useEffect(() => {
    let evalInput = newTask.replaceAll(" ", "");
    if (evalInput.length >= 1 && evalInput != "") {
      setIsDisabled(false);
    } else {
      setIsDisabled(true);
    }
  }, [newTask]);

  const createNewTask = async (
    str: string,
    event?: React.FormEvent<HTMLFormElement>
  ) => {
    event?.preventDefault();
    let response = await createTask(str);
    if (response.itemCreated) {
      setAllTasks([
        ...allTasks,
        { id: response.id, title: str, completed: false },
      ]);
      setNewTask("");
    }
  };

  return (
    <Container style={{ marginBottom: "1vh" }}>
      <Row>
        <Col xs={4}></Col>
        <Col xs={8} className="formHolder">
          <Form onSubmit={(e) => createNewTask(newTask, e)}>
            <Form.Control
              value={newTask}
              id="createTaskInput"
              onChange={(e) => setNewTask(e.target.value)}
            />
            <Button
              variant="success"
              id="createTaskButton"
              disabled={isDisabled}
              type="submit"
            >
              Create
            </Button>
          </Form>
        </Col>
      </Row>
    </Container>
  );
};
