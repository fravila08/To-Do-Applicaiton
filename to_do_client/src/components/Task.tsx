import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { ITask } from "../App";
import Form from "react-bootstrap/Form";
import axios from "axios";
import done from "../assets/done.png";
import pending from "../assets/pending.png";

export interface TaskProps {
  task: ITask;
  selectedTasks: number[];
  setSelectedTasks: (selectedTasks: number[]) => void;
  allTasks: ITask[];
  setAllTasks: (allTasks: ITask[]) => void;
}

export const changeTaskStatus = async (id: number) => {
  try {
    let response = await axios.put(`task/${id}/`);
    return response.data.changed;
  } catch (err) {
    alert(err);
    return false;
  }
};

export const Task: React.FC<TaskProps> = ({
  task,
  setSelectedTasks,
  selectedTasks,
  allTasks,
  setAllTasks,
}) => {
  const changeStatus = async (clicked: boolean, taskToChange: ITask) => {
    let response = await changeTaskStatus(taskToChange["id"]);
    if (response) {
      setAllTasks(allTasks.filter((task) => task !== taskToChange));
      taskToChange.completed = clicked;
      setAllTasks([...allTasks]);
    }
  };

  return (
    <Row className="task">
      <Col>
        <Form.Check
          type="checkbox"
          id={`taskSelectedBtn${task.id}`}
          onChange={(e) =>
            e.target.checked
              ? setSelectedTasks([...selectedTasks, task.id])
              : setSelectedTasks(selectedTasks.filter((id) => id !== task.id))
          }
        />
      </Col>
      <Col id={`task${task.id}`} xs={7} className="taskTitle">
        {task.title}
      </Col>
      <Col className="checkHolder" xs={2}>
        <Form.Check
          id={`taskCheck${task.id}`}
          type="checkbox"
          name="checkbox"
          className="check"
          checked={task.completed}
          onChange={(e) => changeStatus(e.target.checked, task)}
        />
        <Form.Label for="checkbox">
          <img className="checkImg" src={task.completed ? done : pending} />
        </Form.Label>
      </Col>
    </Row>
  );
};
