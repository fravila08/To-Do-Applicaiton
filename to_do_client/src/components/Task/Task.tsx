import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Form from 'react-bootstrap/Form'
import { useEffect, useState } from 'react'
import { Tasks } from '../../App'

interface IdSelectionProps{
    selected: number[];
    setSelected: (selected: number[]) => void;
    task: Tasks;
}

// export function addToSelection(selected:number[],id:number, checked:boolean){
//     console.log("hello")
//     let mySelected=selected
//     if(checked){
//         mySelected.push(Number(id))
//     }
//     else{
//         let newSelected:number[]=[]
//         for(let i=0; i< mySelected.length; i++){
//             if(mySelected[i]!==id){
//                 newSelected.push(mySelected[i])
//             }
//         }
//         mySelected=newSelected
//     }
//     console.log(mySelected,'s')
//     return mySelected
// }


export const Task: React.FC<IdSelectionProps>=({setSelected, selected, task})=>{
    const [mySelected, setMySelected]=useState<number[]>(selected)
    
    useEffect(()=>{
        setMySelected(selected)
    },[selected])


    return(
        <Row style={{backgroundColor:"grey", border:"black 2px solid"}}>
            <Col xs={9}>
                {task.title}
            </Col>
            <Col xs={2}>
                <Form.Check type="checkbox" label="select" 
                onChange={(e)=>{
                    if(e.target.checked){
                    setSelected([...selected, task.id])
                    }
                    else{ setSelected(selected.filter(item=> item !== task.id))}
                    }} 
                
                />
            </Col>
        </Row>
    )
}