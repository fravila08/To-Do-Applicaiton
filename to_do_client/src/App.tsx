import React, {  useEffect, useState } from 'react'
import './App.css'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { Task } from './components/Task'
import axios from 'axios'
import { CsrfToken } from './components/CsrfToken'

export interface TaskInterface {
    id: number;
    Title: string;
    Completed:boolean;
}

export interface ResponseTasks{
    completed: TaskInterface[];
    pending: TaskInterface[];
    biggestID: number;
}

export async function getTasks():Promise<ResponseTasks>{
    let response = await axios.get('allTasks')
    return response.data
}


function App() {
  
  const [compTasks, setCompTasks] : [TaskInterface[] , (compTasks:TaskInterface[])=>void]=useState<TaskInterface[]>([])
  const [pendTasks, setPendTasks] : [TaskInterface[] , (pendTasks:TaskInterface[])=>void]=useState<TaskInterface[]>([])
  
  useEffect(()=>{
      const getResponse=async ()=>{ 
          let response=await getTasks()
          setCompTasks(response.completed)
          setPendTasks(response.pending)
      }
      getResponse()
  },[])
  
  CsrfToken()

  return (
    <Container className="App"> 

      <Row id='Header'>
          <h1>To Do App</h1>
      </Row>

      <Row>

        <Col xs={1}></Col>

        <Col xs={10} className='listHolder'>
          
          <h5 className='listHeader'> Pending </h5>
          
          {pendTasks.map((task)=>(
            <Task 
              task={task} 
            />
          ))}

        </Col>

        <Col xs={1}></Col>

      </Row>

      <Row>

        <Col xs={1}></Col>

        <Col xs={10} className='listHolder'>
          
          <h5 className='listHeader'> Completed </h5>
        
          {compTasks.map((task)=>(
            <Task 
              task={task}
            />
          ))}
      
        </Col>

        <Col xs={1}></Col> 

      </Row>
    </Container>
  )
}

export default App
