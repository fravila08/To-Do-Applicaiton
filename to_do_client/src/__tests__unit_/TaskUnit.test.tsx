import { Task } from "../components/Task";
import { describe, expect, it, vi, Mocked } from "vitest";
import TestRenderer from "react-test-renderer";
import { ITask } from "../App";
import { changeTaskStatus } from "../components/Task";
import axios from "axios";

vi.mock("axios");

describe("Task", () => {
  describe("changeTaskStatus()", () => {
    it("will return if a task's completed value was changed", async () => {
      const mockedAxios = axios as Mocked<typeof axios>;
      mockedAxios.put.mockResolvedValue({
        data: { changed: true },
      });
      const newTasks = await changeTaskStatus(1);
      expect(newTasks).toBeTruthy();
    });
  });
});

describe("Task", () => {
  it("will create and match a snapshot", () => {
    let pendTasks: ITask[] = [];
    const setPendTasks = (newPendTasks: ITask[]) => {
      pendTasks = newPendTasks;
    };
    let compTasks: ITask[] = [];
    const setComptTasks = (other: ITask[]) => {
      compTasks = other;
    };
    const header = TestRenderer.create(
      <Task
        pendingTasks={pendTasks}
        setPendingTasks={setPendTasks}
        completedTasks={compTasks}
        setCompletedTasks={setComptTasks}
        task={{ id: 1, title: "Style", completed: false }}
      />
    );
    expect(header).toMatchSnapshot();
  });
});
