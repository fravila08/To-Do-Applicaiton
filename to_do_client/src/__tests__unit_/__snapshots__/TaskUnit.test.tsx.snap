// Vitest Snapshot v1

exports[`Task > will create and match a snapshot 1`] = `
<div
  className="task row"
  id="taskMaster1"
>
  <div
    className="col-1"
  >
    <div
      className=""
    >
      <input
        className="form-check-input"
        disabled={false}
        id="taskSelectedBtn1"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="taskTitle col-7"
    id="task1"
  >
    Style
  </div>
  <div
    className="checkHolder col-1"
  >
    <div
      className="check"
    >
      <input
        checked={false}
        className="form-check-input"
        disabled={false}
        id="taskCheck1"
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <label
      className="form-label"
      for="checkbox"
    >
      <img
        className="checkImg"
        src="/static/src/assets/pending.png"
      />
    </label>
  </div>
  <div
    className="buttonHolder col-2"
  >
    <button
      className="alterBtn btn btn-danger"
      disabled={false}
      id="deleteBtn1"
      onClick={[Function]}
      type="button"
    >
      <img
        className="buttonImg"
        src="/static/src/assets/trash.png"
      />
    </button>
    <button
      className="alterBtn btn btn-warning"
      disabled={false}
      id="showChangeForm1"
      onClick={[Function]}
      type="button"
    >
      <img
        className="buttonImg"
        src="/static/src/assets/edit.png"
      />
    </button>
  </div>
</div>
`;
